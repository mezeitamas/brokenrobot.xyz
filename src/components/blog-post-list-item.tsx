import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { Link } from 'gatsby';

type BlogPostListItemProps = {
    title: string;
    excerpt: string;
    published: string;
    slug: string;
};

const BlogPostListItem: FunctionComponent<BlogPostListItemProps> = ({
    title,
    excerpt,
    published,
    slug
}): ReactElement => {
    return (
        <article
            className="flex flex-col items-start justify-between"
            key={slug}
        >
            <header>
                <h3>
                    <Link to={`/blog/${slug}/`}>{title}</Link>
                </h3>

                <time
                    dateTime={published}
                    className="text-sm text-gray-600"
                >
                    {published}
                </time>

                <p>{excerpt}</p>
            </header>
        </article>
    );
};

export { BlogPostListItem };
