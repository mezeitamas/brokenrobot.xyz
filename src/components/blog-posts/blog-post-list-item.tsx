import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { Link } from 'gatsby';

type BlogPostListItemProps = {
    title: string;
    excerpt: string;
    published: string;
    publishedFormatted: string;
    slug: string;
    HeadingType: keyof JSX.IntrinsicElements;
};

const BlogPostListItem: FunctionComponent<BlogPostListItemProps> = ({
    title,
    excerpt,
    published,
    publishedFormatted,
    slug,
    HeadingType
}): ReactElement => {
    return (
        <article
            className="flex flex-col items-start justify-between"
            key={slug}
        >
            <header>
                <HeadingType>
                    <Link to={`/blog/${slug}/`}>{title}</Link>
                </HeadingType>

                <time dateTime={published}>{publishedFormatted}</time>

                <p>{excerpt}</p>
            </header>
        </article>
    );
};

export { BlogPostListItem };

export type { BlogPostListItemProps };
