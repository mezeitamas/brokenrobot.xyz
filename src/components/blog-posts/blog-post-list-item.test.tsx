import React from 'react';

import { create } from 'react-test-renderer';

import { BlogPostListItem } from './blog-post-list-item';
import type { BlogPostListItemProps } from './blog-post-list-item';

describe('<BlogPostListItem />', () => {
    test('should render with h2 heading', () => {
        // Given
        const props: BlogPostListItemProps = {
            title: 'Advanced static website hosting with Amazon S3 and CloudFront',
            excerpt:
                'Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...',
            published: '2023-05-29T18:00:00.000Z',
            publishedFormatted: 'May 29, 2023',
            slug: 'advanced-static-website-hosting-with-amazon-s3-and-cloudfront',
            HeadingType: 'h2'
        };

        // When
        const component = create(<BlogPostListItem {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });

    test('should render with h3 heading', () => {
        // Given
        const props: BlogPostListItemProps = {
            title: 'Advanced static website hosting with Amazon S3 and CloudFront',
            excerpt:
                'Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...',
            published: '2023-05-29T18:00:00.000Z',
            publishedFormatted: 'May 29, 2023',
            slug: 'advanced-static-website-hosting-with-amazon-s3-and-cloudfront',
            HeadingType: 'h3'
        };

        // When
        const component = create(<BlogPostListItem {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
