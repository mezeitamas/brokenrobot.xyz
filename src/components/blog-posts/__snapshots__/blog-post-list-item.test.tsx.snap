// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostListItem /> should render with h2 heading 1`] = `
<article
  className="flex flex-col items-start justify-between"
>
  <header>
    <h2>
      <a
        href="/blog/advanced-static-website-hosting-with-amazon-s3-and-cloudfront/"
      >
        Advanced static website hosting with Amazon S3 and CloudFront
      </a>
    </h2>
    <time
      dateTime="2023-05-29T18:00:00.000Z"
    >
      May 29, 2023
    </time>
    <p>
      Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...
    </p>
  </header>
</article>
`;

exports[`<BlogPostListItem /> should render with h3 heading 1`] = `
<article
  className="flex flex-col items-start justify-between"
>
  <header>
    <h3>
      <a
        href="/blog/advanced-static-website-hosting-with-amazon-s3-and-cloudfront/"
      >
        Advanced static website hosting with Amazon S3 and CloudFront
      </a>
    </h3>
    <time
      dateTime="2023-05-29T18:00:00.000Z"
    >
      May 29, 2023
    </time>
    <p>
      Today we will dive deep into the world of hosting a static website on Amazon S3 using CloudFront. If you’re looking for a secure, reliable, scalable, cost effective and performant solution, you’ve come to the right place. In this blog post, we’ll...
    </p>
  </header>
</article>
`;
